packages/replay-next/src/suspense/PointStackCache.ts

```diff
@@ -0,0 +1,37 @@
+import { ExecutionPoint, PointStackFrame } from "@replayio/protocol";
+import { IntervalCache, createIntervalCache } from "suspense";
+
+import { ReplayClientInterface } from "shared/client/types";
+
+import { updateMappedLocation } from "./PauseCache";
+import { sourcesByIdCache } from "./SourcesCache";
+
+export const pointStackCache: IntervalCache<
+  number,
+  [replayClient: ReplayClientInterface, point: ExecutionPoint],
+  PointStackFrame & { index: number }
+> = createIntervalCache({
+  debugLabel: "pointStackCache",
+  getPointForValue: pointStackFrame => pointStackFrame.index,
+  getKey: (client, point) => point,
+  load: async (start, end, client, point) => {
+    const pointStack = await client.getPointStack(point, end + 1);
+    const sources = await sourcesByIdCache.readAsync(client);
+    return pointStack.slice(start).map((frame, index) => {
+      updateMappedLocation(sources, frame.functionLocation);
+      if (frame.point.frame) {
+        updateMappedLocation(sources, frame.point.frame);
+      }
+      return { ...frame, index: index + start };
+    });
+  },
+});
+
+export async function getPointDescriptionForFrame(
```

Comment:
I don't love these meta cache methods because we end up being able to use them with other utilities exported by the "suspense" package (the hooks and stuff). I guess in this case it doesn't matter much since the only things using this cache are other async functions, never React components.